packages/dashboard/src/vaadin-dashboard-item-mixin.js

```diff
@@ -377,4 +395,33 @@ export const DashboardItemMixin = (superClass) =>
       }
       this.dispatchEvent(new CustomEvent('item-resize-mode-changed', { bubbles: true, detail: { value: resizeMode } }));
     }
+
+    /** @private */
+    __setupHeadingLevelObserver() {
+      this.__removeHeadingLevelObserver();
+      const parentLayout = getParentLayout(this);
+      if (parentLayout) {
+        this.__headingLevelObserver = new MutationObserver(() => this.__updateRootHeadingLevel());
```

Comment:
Should we just fire an event when the heading level changes? MIght be a bit more light-weight than creating a mutation observer per item instance.

Reply:
@ugur-vaadin Did you plan to address this?